src/app/core/auth/auth.reducer.ts

```diff
@@ -49,8 +48,8 @@ export interface AuthState {
   // true when refreshing token
   refreshing?: boolean;
 
-  // the authenticated user
-  user?: EPerson;
+  // the authenticated user's id
+  userId?: string;
```

Comment:
The idea is that every object is stored exactly once inside the store. If you don't, and the object changes, you need to keep all places where it is stored in sync. 

In this case, the when the active user changed, the dropdown didn't change with it because the auth section of the store had a separate copy of the user object. That's why I asked @Atmire-Kristof to use only the id instead.

Reply:
Correct me if i'm wrong but the userId in the state is updated with current user id, even if it's an impersonated one. After impersonating an eperson the page is reload so the store is up to date with the current user. the dropdown menu show who is the current user, but unlike before it make the request to retrieve the eperson data instead to use the object in the store. So why don't retrieve the user data once at login as before if the store seems to contain always the current user ?

Reply:
It didn't work that way before Kristof changed it. The user in the auth state, and in the dropdown would always be the admin, not the impersonated user, he made this change specifically to address that.

As for the reason an extra EPerson request is sent, that seems to be due to the projections changes on the rest side. The EPerson isn't embedded in the status request unless you add an `?embed=eperson` param. Embedding it could be a slight performance increase, but it is unrelated to this PR.

Reply:
As I can see in the code it should be enough to change here https://github.com/DSpace/dspace-angular/blob/3894b3615d4d788c9d760c965ca87c367655d0d8/src/app/core/auth/auth.effects.ts#L109